airflow-core/src/airflow/api_fastapi/common/parameters.py

```diff
@@ -185,6 +185,53 @@ def depends(cls, *args: Any, **kwargs: Any) -> Self:
         raise NotImplementedError("Use search_param_factory instead , depends is not implemented.")
 
 
+class QueryTaskInstanceTaskGroupFilter(BaseParam[str]):
+    """Task group filter - returns all tasks in the specified group."""
+
+    def __init__(self, dag=None, skip_none: bool = True):
+        super().__init__(skip_none=skip_none)
+        self._dag: None | Any = dag
+
+    @property
+    def dag(self) -> None | Any:
+        return self._dag
+
+    @dag.setter
+    def dag(self, value: None | Any) -> None:
+        self._dag = value
```

Comment:
The `SerializedDAG` can be imported from `airflow.serialization.serialized_objects` under `TYPE_CHECKING` block.

```suggestion
        self._dag: None | SerializedDAG = dag

    @property
    def dag(self) -> None | SerializedDAG:
        return self._dag

    @dag.setter
    def dag(self, value: SerializedDAG) -> None:
        self._dag = value
```